src/test/java/dk/kb/util/yaml/YAMLTest.java

```diff
@@ -528,4 +528,41 @@ private void assertExtrapolated(String path, String expected, String message) th
         String actual = yaml.getString(path);
         assertEquals(expected, actual, message);
     }
+
+    @Test
+    public void testPathSubstitute() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("yaml/path_substitution.yaml");
+        yaml.setExtrapolate(true);
+
+        assertEquals("foo", yaml.get(".lower.bar"), "Basic path substitution should work");
+        assertEquals("boom", yaml.get(".fallback.bar"), "Path substitution with default value should work");
+        assertEquals("foo", yaml.get(".mixing.bar"), "Path substitution with default value should work");
+    }
+
+    @Test
+    public void testConditionalIndex() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("nested_maps.yml");
+        assertEquals("boom", yaml.get("conditionalmap.[default=true].foo"));
+    }
+
+    @Test
+    public void testReverseConditionalIndex() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("nested_maps.yml");
+        assertEquals("bar", yaml.get("conditionalmap.[default!=true].foo"));
+    }
+
+    @Test
+    public void testConditionalIndexFlat() throws IOException {
+        YAML yaml = YAML.resolveLayeredConfigs("nested_maps.yml");
+        assertEquals("boom", yaml.get("conditionalflat.[default=true].foo"));
+    }
+
```

Comment:
Same as above, but for the conditionalflatmap

Reply:
Same answer as above :-)